src/vaults/AsyncRequestManager.sol

```diff
@@ -524,9 +524,11 @@ contract AsyncRequestManager is BaseRequestManager, IAsyncRequestManager {
         VaultDetails memory vaultDetails = spoke.vaultDetails(vault_);
         address shareToken = vault_.share();
 
-        return PricingLib.assetToShareAmount(
-            shareToken, vaultDetails.asset, vaultDetails.tokenId, assets, d18(priceAssetPerShare.toUint128()), rounding
-        );
+        return priceAssetPerShare == 0
```

Comment:
What was the reason we're not using `D18` here?

Reply:
Good point! Fixed in [5a8d733](https://github.com/centrifuge/protocol-v3/pull/440/commits/5a8d7331e1f788875003e3944032581791ed3df7)

Reply:
Thanks for doing this!